### ISSUE TITLE
document an MSRV

### ISSUE THREAD

**Issue by zachschuermann** [delta-io/delta-kernel-rs#298]
I would assume we implicitly have an msrv, probably need to actually document + test it? Maybe more critical after a 1.x release

**Comment by hntd187** [comment]
A tool, cargo-msrv (a tool that just installs and tries to build on every version of rust) determined currently an MSRV would be around 1.77.2. This is probably a reasonable place to start, most of the reasons it looks like we don't build on older versions are build related things we do to enable/disable doc-comments and such.

**Comment by zachschuermann** [comment]
That's actually a bit surprising to me - figured we would be able to build older.. 1.77 is only 4-5 months old I think..?

**Comment by hntd187** [comment]
It's about 5 months ago, https://blog.rust-lang.org/2024/04/09/Rust-1.77.2.html

Either way, the builds point to the use of some build features we have like `#![cfg_attr(all(doc, NIGHTLY_CHANNEL), feature(doc_auto_cfg))]` not being able to be understood. 

```
│ error: unsupported output in build script of `delta_kernel v0.2.0 (C:\Users\shcar\IdeaProjects\delta-kernel-rs\kernel)`: `cargo::rustc-check-cfg=cfg(NIGHTLY_CHANNEL)`                                                                                                                                           │
│ Found a `cargo::key=value` build directive which is reserved for future use.                                                                                                                                                                                                                                     │
│ Either change the directive to `cargo:key=value` syntax (note the single `:`) or upgrade your version of Rust.                                                                                                                                                                                                   │
│ See https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script for more information about build script outputs.                                                                                                                                                                    │
│ warning: build failed, waiting for other jobs to finish...  
```

**Comment by zachschuermann** [comment]
resolved in #504
